src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -50,6 +50,9 @@ public class ProjectRestoreMetadata : IEquatable<ProjectRestoreMetadata>
         /// </summary>
         public string PackagesPath { get; set; }
 
+        public string CacheFilePath {
+            get { return OutputPath + "project.nuget.json"; }  }
```

Comment:
I had a discussion with @rohit21agrawal, @joelverhagen and @mishra14 and they all share the opinion that it should live in the obj folder for both PJ + Pack Ref and be named "projectName.nuget.cache"